pyflagser/flagser.py

```diff
@@ -1,11 +1,127 @@
+        meaning that the filtration value of every simplex of dimension at
+        least two should evaluate to a value that is at least the maximal value
+        of the filtration values of its containing edges. For performance
+        reasons, this is not checked automatically.  Possible values are:
+        ['dimension', 'zero', 'max', 'max3', 'max_plus_one', 'product', 'sum',
+        'pmean', 'pmoment', 'remove_edges', 'vertex_degree']
+
+    coeff : int, optional, default: ``2``
+        Compute homology with coefficients in the prime field
+        :math:`\\mathbb{F}_p = \\{ 0, \\ldots, p - 1 \\}` where
+        :math:`p` equals `coeff`.
+
+    approximation : int or None, optional, default: ``None``
+        Skip all cells creating columns in the reduction matrix with more than
+        this number of entries. Use this for hard problems; a good value is
+        often ``100,000``. Increase for higher precision, decrease for faster
+        computation. A negative value computes highest possible precision. If
+        ``None``, no approximation is used.
+
+    Returns
+    -------
+    out : dict of list
+        A dictionary holding the results of the flagser computation. Its
+        key-value pairs are as follows:
+
+        - ``'dgms'``: list of ndarray of shape ``(n_pairs, 2)``
+          A list of persistence diagrams, one for each dimension greater
+          than or equal than `min_dimension` and less than `max_dimension`.
+          Each diagram is an ndarray of size (n_pairs, 2) with the first
+          column representing the birth time and the second column
+          representing the death time of each pair.
+        - ``'cell_count'``: list of int
+          Cell count per dimension greater than or equal than
+          `min_dimension` and less than `max_dimension`.
+        - ``'betti'``: list of int
+          Betti number per dimension greater than or equal than
+          `min_dimension` and less than `max_dimension`.
+        - ``'euler'``: int
+          Euler characteristic per dimension greater than or equal than
+          `min_dimension` and less than `max_dimension`.
+
+    Notes
+    -----
+    For more details, please refer to the `flagser documentation \
+    <https://github.com/luetge/flagser/blob/master/docs/\
+    documentation_flagser.pdf>`_.
+
+    """
+    # Handle default parameters
+    if max_dimension == np.inf:
+        _max_dimension = -1
+    else:
+        _max_dimension = max_dimension
+
+    if approximation is None:
+        _approximation = -1
+    else:
+        _approximation = approximation
+
+    if filtration not in implemented_filtrations:
+        raise ValueError("Filtration not recongnized. Available filtrations "
+                         "are ", implemented_filtrations)
+
+    # Extract vertices and edges weights
+    vertices, edges = _extract_static_weights(flag_matrix)
+    edges = np.hstack([edges, np.ones(edges[:, [0]].shape, dtype=np.int)])
```

Comment:
Are you saying that if no weight is passed then C++ assumes this weight is `1.`? And that since we are now passing an array of `True` for `vertices`, `True` is also converted to `1.`, and hence vertices and edges have the same weights in the internal computation?

Reply:
No, I am saying that if there is no weight, we need to tell C++ flagser that we mean weight ``1`` for both vertices and edges.